src/sentry/static/sentry/app/components/dropdownMenu.jsx

```diff
@@ -0,0 +1,227 @@
+      return;
+
+    if (typeof onClickOutside === 'function') {
+      onClickOutside(e);
+    }
+
+    this.handleClose(e);
+  };
+
+  // Callback function from <DropdownMenu> to see if we should close menu
+  shouldIgnoreClickOutside = e => {
+    let {shouldIgnoreClickOutside} = this.props;
+    if (this.dropdownActor.contains(e.target)) return true;
+    if (typeof shouldIgnoreClickOutside === 'function') {
+      return shouldIgnoreClickOutside(e);
+    }
+
+    return false;
+  };
+
+  // Opens dropdown menu
+  handleOpen = e => {
+    let {onOpen, isOpen} = this.props;
+    let isControlled = typeof isOpen !== 'undefined';
+    if (!isControlled) {
+      this.setState({
+        isOpen: true,
+      });
+    }
+
+    if (typeof onOpen === 'function') {
+      onOpen(e);
+    }
+  };
+
+  // Decide whether dropdown should be closed when mouse leaves element
+  handleMouseLeave = e => {
+    const toElement = e.toElement || e.relatedTarget;
+
+    try {
+      if (this.dropdownMenu && !this.dropdownMenu.contains(toElement)) {
+        this.handleClose(e);
+      }
+    } catch (err) {
+      Raven.captureException(err, {
+        event: e,
+        toElement: e.toElement,
+        relatedTarget: e.relatedTarget,
+      });
+    }
+  };
+
+  // Closes dropdown menu
+  handleClose = e => {
+    let {onClose, isOpen} = this.props;
+    let isControlled = typeof isOpen !== 'undefined';
+
+    if (!isControlled) {
+      this.setState({isOpen: false});
+    }
+
+    if (typeof onClose === 'function') {
+      onClose(e);
+    }
+  };
+
+  // When dropdown menu is displayed and mounted to DOM,
+  // bind a click handler to `document` to listen for clicks outside of
+  // this component and close menu if so
+  handleMenuMount = ref => {
+    this.dropdownMenu = ref;
+
+    if (this.dropdownMenu) {
+      // 3rd arg = useCapture = so event capturing vs event bubbling
+      document.addEventListener('click', this.checkClickOutside, true);
+    } else {
+      document.removeEventListener('click', this.checkClickOutside, true);
+    }
+  };
+
+  handleToggle = e => {
+    if (this.isOpen()) {
+      this.handleClose(e);
+    } else {
+      this.handleOpen(e);
+    }
+  };
+
+  // Control whether we should hide dropdown menu when it is clicked
+  handleDropdownMenuClick = e => {
+    if (this.props.keepMenuOpen) return;
+
+    this.handleClose(e);
+  };
+
+  getRootProps = props => props;
+
+  getActorProps = ({onClick, ...props} = {}) => {
+    let {isNestedDropdown} = this.props;
+    return {
+      ...props,
+      ref: ref => (this.dropdownActor = ref),
+      onMouseEnter: (...args) => {
+        if (!isNestedDropdown) return;
+        this.handleOpen(...args);
+      },
+
+      onMouseLeave: (...args) => {
+        if (!isNestedDropdown) return;
+        this.handleMouseLeave(...args);
+      },
+      onClick: (...args) => {
+        if (!isNestedDropdown) {
```

Comment:
I think we should toggle regardless of whether the dropdown is nested or not. Otherwise dropdowns might stay open after we're done interacting with them.